### ISSUE TITLE
Simulate basic aspects of a nation

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1096]
**Is your feature request related to a problem? Please describe.**
- Currently, mars-sim does not provide any mechanics on how a country impacts its Moon or Mars settlement. 

**Describe the rationale**
- To better handle how the policy and economic prospect of a nation would influence the planning aspects and mission operations of its space agency, we could provide the basic economic simulation of the involving nations.

**Describe the solution you desire to see**
1. Add brand new national attributes in each existing country's xml. e.g. `country_india.xml`
2. Rename `PersonNameSpecConfig` to `NationSpecConfig`
3. Rename `PersonNameSpec` to `NationSpec`

**Specifications  (please complete the following information):**
 -pre 3.7.0 build 8788

**Additional context**
1. Add new attributes in `ReportingAuthority` to track how economic or financial resources are being utilized
2. Add a new nation and sponsor UI elements (a panel such as `UnitWindow`) for players to understand and interact with the internal dynamics of one's nation and sponsor
3. Work with #906

**Comment by bevans2000** [comment]
Sounds like you should create a Github Project to track all these related issues

**Comment by bevans2000** [comment]
You could expand this to move the Settlement & Rover names from ```ReportingAuthority``` to the new ```NationSpec```. A new Rover/Settlement name could be picked by choosing one from the underlying Nations.
The benefit is that the NationSpec is lazy loaded and only in memory when needed; this means all those String for vehicles & settlement would only reside in memory when we are creating Settlements via the Settlement Builder.

**Comment by mokun** [comment]
>  Github Project to track all these related issues

Done !

**Comment by mokun** [comment]
> A new Rover/Settlement name could be picked by choosing one from the underlying Nations.

Right.

> all those String for vehicles & settlement would only reside in memory when we are creating Settlements via the Settlement Builder.

I thought they are already lazy-loaded. Don't they ? 

If I start a 'Blue Origin' settlement template, then it would load the country 'United States' only. 

In that case, it never actually need to load China's portion in governance.xml into the memory, right ?

**Comment by bevans2000** [comment]
> > A new Rover/Settlement name could be picked by choosing one from the underlying Nations.
> 
> Right.
> 
> > all those String for vehicles & settlement would only reside in memory when we are creating Settlements via the Settlement Builder.
> 
> I thought they are already lazy-loaded. Don't they ?
> 
> If I start a 'Blue Origin' settlement template, then it would load the country 'United States' only.
> 
> In that case, it never actually need to load China's portion in governance.xml into the memory, right ?

Correct, only the required Country is loaded when needed but Settlement and Vehicle names are held under the Reporting Authority that is always in memory.
